include/alpaka/mem/buf/omp5/Copy.hpp

```diff
@@ -0,0 +1,545 @@
+                        }
+                    };
+
+                    //#############################################################################
+                    //! The Omp5 memory copy trait.
+                    template<
+                        typename TDim,
+                        typename TViewDst,
+                        typename TViewSrc,
+                        typename TExtent>
+                    struct TaskCopyOmp5
+                    {
+                        static_assert(
+                            !std::is_const<TViewDst>::value,
+                            "The destination view can not be const!");
+
+                        static_assert(
+                            dim::Dim<TViewSrc>::value == TDim::value,
+                            "The source view is required to have dimensionality TDim!");
+                        static_assert(
+                            dim::Dim<TViewDst>::value == dim::Dim<TViewSrc>::value,
+                            "The source and the destination view are required to have the same dimensionality!");
+                        static_assert(
+                            dim::Dim<TViewDst>::value == dim::Dim<TExtent>::value,
+                            "The views and the extent are required to have the same dimensionality!");
+                        // TODO: Maybe check for Idx of TViewDst and TViewSrc to have greater or equal range than TExtent.
+                        static_assert(
+                            std::is_same<elem::Elem<TViewDst>, typename std::remove_const<elem::Elem<TViewSrc>>::type>::value,
+                            "The source and the destination view are required to have the same element type!");
+
+                        using Idx = idx::Idx<TExtent>;
+
+                        //-----------------------------------------------------------------------------
+                        ALPAKA_FN_HOST TaskCopyOmp5(
+                            TViewDst & viewDst,
+                            TViewSrc const & viewSrc,
+                            TExtent const & extent,
+                            int const & iDstDevice,
+                            int const & iSrcDevice) :
+                                m_iDstDevice(iDstDevice),
+                                m_iSrcDevice(iSrcDevice),
+                                m_extent(VecFromDimTrait<
+                                        TDim, size_t, TExtent,
+                                        MyGetExtent>::vecFromDimTrait(extent)),
+                                m_dstPitchBytes(VecFromDimTrait<
+                                        TDim, size_t, TViewDst,
+                                        MyGetPitch>::vecFromDimTrait(viewDst)),
+                                m_srcPitchBytes(VecFromDimTrait<
+                                        TDim, size_t, TViewSrc,
+                                        MyGetPitch>::vecFromDimTrait(viewSrc)),
```

Comment:
What is the difference to the `mem::view::getPitchBytesVec(viewSrc), that is used in ` TaskCopyCpuBase` (and `getExtentVec` for above)?
`

Reply:
This version is casting to a `Vec<size_t>`. The class stores the extents with this type because they are directly used as arguments to the omp runtime functions, which expect arrays of `size_t`. An alternative to this may be to add a cast operator to `Vec<..>`.

Reply:
You can find it here: https://github.com/alpaka-group/alpaka/blob/d1f7047ae2116cc6a83674ce352adabbb263ad04/test/unit/vec/src/VecTest.cpp#L91-L118

Reply:
Thanks; I put it in.